src/Cardano/Wallet.hs

```diff
@@ -95,23 +103,25 @@ mkWalletLayer db network = WalletLayer
                 mkAddressPool (publicKey accXPrv) (gap w) ExternalChain []
         let intPool =
                 mkAddressPool (publicKey accXPrv) minBound InternalChain []
-        let wallet =
-                initWallet $ SeqState (extPool, intPool)
+        let wallet = initWallet $ SeqState
+                { externalPool = extPool
+                , internalPool = intPool
+                }
         -- FIXME Compute the wallet id deterministically from the seed
         let wid = WalletId (read "00000000-0000-0000-0000-000000000000")
-        lift (readCheckpoints db (PrimaryKey wid)) >>= \case
+        liftIO (readCheckpoints db (PrimaryKey wid)) >>= \case
             Nothing -> do
-                lift $ putCheckpoints db (PrimaryKey wid) (wallet :| [])
+                liftIO $ putCheckpoints db (PrimaryKey wid) (wallet :| [])
                 return wid
             Just _ ->
                 throwE $ ErrCreateWalletIdAlreadyExists wid
-    , getWallet = \wid -> lift (readCheckpoints db (PrimaryKey wid)) >>= \case
+    , readWallet = \wid -> liftIO (readCheckpoints db (PrimaryKey wid)) >>= \case
         Nothing ->
-            throwE $ ErrGetWalletNotFound wid
+            throwE $ ErrReadWalletNotFound wid
         Just (w :| _) ->
-            return w
+            return (w, error "FIXME: store and retrieve wallet metadata")
```

Comment:
Will do in another PR. Thanks for Haskell lazyness magic, tests still pass :sweat_smile:  ...

Reply:
:+1: